### ISSUE TITLE
[BUG] Error when site name contains space for Tableau connector

### ISSUE THREAD

**Issue by ealtuna** [GoogleCloudPlatform/datacatalog-connectors-bi#43]
**What happened**:

I am running the connector to ingest the metadata from a Tableau server on the cloud to Google Catalog. The name of the site in question contain an space in the name (like: My site). When providing this name to the connector is in the format myname.

Google Catalog fails because it does not support spaces: 

```
google.api_core.exceptions.InvalidArgument: 400 
"https://us-east-1.online.tableau.com/#/site/My Site/workbooks/594378" is an invalid value for CreateEntryRequest.entry.linked_resource name. 
It must: contain only letters, numbers, periods, colons, slashs, underscores, dashes and hashes and be at most 200  bytes long when encoded in UTF-8.
```

**What you expected to happen**:

Metadata should be replicated to Google Data Catalog

**How to reproduce it (as minimally and precisely as possible)**:

Use a site where the display name contains an space. Creates a workbook inside and run the connector.

**Anything else we need to know?**:

Looking at the code I see an issue in the method `__format_site_content_url` where `'/site/{site_name}'` is used to build the site url part. This is not going to yield a valid url in many cases.

**Comment by ricardolsmendes** [comment]
Hi @ealtuna, thanks for the detailed bug description! 👏🏼 
`__format_site_content_url` can be improved to replace spaces with `%20` and we would have well-formatted URLs, but based on the error message you reported I still think the entry would be rejected by Data Catalog's API due to the `%` char. If so, we will need to file a bug to the product team before fixing the Tableau connector.

Would you like to run such a test and confirm my guess?

**Comment by ealtuna** [comment]
@ricardolsmendes the thing is, even if you escape the space the URL will be invalid. In the URL what Tableau uses is not the site name (display) `My Site` but the actual site name (in my example `mysite`). The problem is I was playing with Tableau's graph API and the actual site name is not returned. At least this is the behavior in the cloud version I am using. Another possible solution (but it does not generate a valid url) will be to use the field uri from the site (something like `/sites/12345`).

**Comment by ricardolsmendes** [comment]
Got it, @ealtuna! Looks like `contentUrl` is the correct field to be used in this case. It's available in the REST API (https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref.htm#query_site) but not in the Metadata API (https://help.tableau.com/current/api/metadata_api/en-us/reference/tableausite.doc.html). Could you query the REST API to confirm?

**Comment by ealtuna** [comment]
@ricardolsmendes confirmed in my case also I am getting the correct value in the field `contentUrl="mysite"`

**Comment by ricardolsmendes** [comment]
Thanks for your help, @ealtuna.
I'm thinking about a quick workaround that consists of replacing spaces with `_` ASAP. Pros: it will avoid breaking the ingestion flow when there are sites with spaces in their names. Cons: there will be invalid URLs in some cases until we have a more robust solution in place. Does it make sense for you?

About the definitive fix, I have some ideas, but need more time to code and test them. If you happen to be interested in contributing a fix, we can discuss the solution. Otherwise, no worries, I'll work on it in a few weeks.

Kind regards,

**Comment by ealtuna** [comment]
Thank you for the fast response. The workaround is good enough as a temporal fix.

I would love to contribute but my time is limited at this time.

**Comment by ricardolsmendes** [comment]
No problem, we'll get to this as soon as we can.

**Comment by ricardolsmendes** [comment]
Reopening the issue because although we have a workaround, the definitive fix is not in place yet.

**Comment by ricardolsmendes** [comment]
Hi @ealtuna, the workaround has been published: https://pypi.org/project/google-datacatalog-tableau-connector/0.6.1/. The bug fix is coming soon.

**Comment by ricardolsmendes** [comment]
Hey, @ealtuna, the bug has been fixed in version `0.6.2`: https://pypi.org/project/google-datacatalog-tableau-connector/0.6.2.
